package/MDAnalysis/analysis/rdf.py

```diff
@@ -261,6 +272,13 @@ def _single_frame(self):
             mask = np.where(idxA != idxB)[0]
             dist = dist[mask]
 
+        if self.exclude_same:
+            # Ignore distances between atoms in the same residue
+            resixA = self.g1.resindices[pairs[:, 0]]
```

Comment:
You need to change the attr hit here as currently only `resindices`?

Reply:
Good catch